openpype/modules/deadline/plugins/publish/submit_publish_job.py

```diff
@@ -933,15 +933,18 @@ def process(self, instance):
 
         self.log.info(data.get("expectedFiles"))
 
-        additional_data = {
-            "renderProducts": instance.data["renderProducts"],
-            "colorspaceConfig": instance.data["colorspaceConfig"],
-            "display": instance.data["colorspaceDisplay"],
-            "view": instance.data["colorspaceView"],
-            "colorspaceTemplate": instance.data["colorspaceConfig"].replace(
-                str(context.data["anatomy"].roots["work"]), "{root[work]}"
-            )
-        }
+        additional_data = {}
+        if pyblish.api.current_host() == "maya":
+            config = instance.data["colorspaceConfig"]
+            additional_data = {
+                "renderProducts": instance.data["renderProducts"],
+                "colorspaceConfig": instance.data["colorspaceConfig"],
+                "display": instance.data["colorspaceDisplay"],
+                "view": instance.data["colorspaceView"],
+                "colorspaceTemplate": config.replace(
+                    str(context.data["anatomy"].roots["work"]), "{root[work]}"
+                )
```

Comment:
**Not an issue introduced by this PR**

What if the root is `/mnt/P` and the path had `/mnt/P/projects/mnt/P/files`? This should only replace the actual exact beginning? What if the root was just `/projects` or `/share` - no idea what someone comes up with on unix.
 
Also, what's ensuring the path dividers `/` or `\` are the same - and path case sensitivity on some platforms?

I feel like we might need a dedicated safer method here?

Should this instead be using [`anatomy.find_root_template_from_path`](find_root_template_from_path)? 

```python
anatomy = context.data["anatomy"]
result, template = anatomy.find_root_template_from_path(config)
```

Reply:
> Should this instead be using [anatomy.find_root_template_from_path](https://github.com/ynput/OpenPype/pull/find_root_template_from_path)?

That is one point, yes. I agree that the case you've described might happen. 

> colorspaceTemplate only exists in this code - why not directly acquire the data in the way we're [setting them](https://github.com/ynput/OpenPype/blob/f037074747d040caf12b9af565650a8ae6d32bf3/openpype/modules/deadline/plugins/publish/submit_publish_job.py#L559-L567)?

Perhpas this should be moved to the code you are suggesting, but there was actually created abstracted functionality in here which could be used. There is only missing Display and View input.

Reply:
@jakubjezek001 @BigRoy could we move this to a dedicated issue to tackle outside of this PR?

Reply:
Fine with me.